### PR TITLE
Ecalpedestal pcl improved

### PR DESCRIPTION
**Pull request by argiro**
Adding  flexibility:

1. dynamic booking of histograms
2. configurable number of presamples
3. option to exclude signal frames
4. option to avoid payload creation when a configurable fraction of channels has large pedestal variations

changed default on channels to be excluded

### CODE REVIEW

**Comment by franzoni** [comment]
please test

**Comment by argiro** [comment]
can this be reviewed and hopefully  integrated soon ? We are looking forward to a backport to be able to deploy the PCL in data taking asap.

thanks

File: Calibration/EcalCalibAlgos/interface/ECALpedestalPCLworker.h [hunk 1]
```diff
@@ -53,9 +53,16 @@ class ECALpedestalPCLworker : public  DQMEDAnalyzer {
     std::vector<MonitorElement *> meEB_;
     std::vector<MonitorElement *> meEE_;
 
-    static const int kPedestalSamples = 3;
-    static const int kThreshold = 10; // threshold (in adc count) above which we'll assume 
-                                      // there's signal and not just pedestal
+    uint32_t pedestalSamples_ ; // number of presamples to be used for pedestal determination    
+    bool     checkSignal_;      // avoid frames containing a signal
+    uint32_t  sThreshold_ ;     // if checkSignal = true threshold (in adc count) above which we'll assume 
+                                // there's signal and not just pedestal
```
Comment (franzoni) [inline Calibration/EcalCalibAlgos/interface/ECALpedestalPCLworker.h]: hello @argiro 

With the noise being significantly different between EB and EE (factor 2-3),
and somewhat |eta| dependent in EE
the significance to have "signal" in EB and EE dataframes
could be set here in relative terms in units of of sigma's of the noise
and the check carried out against Numsigna's x sigma_in_db

**Comment by franzoni** [comment]
no unexpected changes in the relval-based workflows 
can be signed once we agree on the matter I've commented about above

**Comment by argiro** [comment]
I've updated the branch with separate threshold parameters for EB and EE

**Comment by argiro** [comment]
btw, I don't think this feature for signal detection will ever be used. After discussion with Ecal we agreed on ignoring the presence/absence of signal and take the first two samples, which are guaranteed to be unbiased pedestal

**Comment by franzoni** [comment]
+1

**Comment by franzoni** [comment]
please test

**Comment by franzoni** [comment]
@davidlange6 
We world much benefit from having this pr integrated ASAP
to commission the workflow

As said above, it targets pcl dependent, for which there is a DataProcessing  scenario
and has no interplay with standard MC or data sequences

Thanks!

File: Calibration/EcalCalibAlgos/src/ECALpedestalPCLHarvester.cc [hunk 2]
```diff
@@ -141,3 +152,39 @@ bool ECALpedestalPCLHarvester::checkStatusCode(const DetId& id){
...
+    uint32_t nAnomaliesEB =0;
+    uint32_t nAnomaliesEE =0;
+
+    for (uint16_t i =0; i< EBDetId::kSizeForDenseIndexing; ++i) {
+       
+        DetId id = EBDetId::detIdFromDenseIndex(i);
+        const EcalPedestal& newped=* newPedestals.find(id.rawId());   
+        const EcalPedestal& oldped=* oldPedestals.find(id.rawId());
+
+        if  (abs(newped.mean_x12 -oldped.mean_x12)  >  nSigma_ * oldped.rms_x12) nAnomaliesEB++;
```
Comment (davidlange6) [inline Calibration/EcalCalibAlgos/src/ECALpedestalPCLHarvester.cc]: hi @argiro - please use std:abs.

**Comment by argiro** [comment]
done

**Comment by franzoni** [comment]
+1

**Comment by franzoni** [comment]
please test

**Comment by davidlange6** [comment]
+1

**Pull request merged**